api/jobs/gears.py

```diff
@@ -119,12 +132,17 @@ def insert_gear(doc):
     if doc.get("invocation-schema"):
         del(doc["invocation-schema"])
 
+    now = datetime.datetime.utcnow()
+
+    doc['created']  = now
+    doc['modified'] = now
+
     config.db.gears.insert(doc)
```

Comment:
I could hide the typing in the map & mongo functions, and do a check type on init?